UnsplashApiApp/Helpers/UIImageView+AsyncDownload.swift

```diff
@@ -11,29 +11,15 @@ import Foundation
 
 extension UIImageView {
     func imageFromServerURL(_ imageURL: String, placeHolder: UIImage?) {
-        func setPlaceHolder() {
-            DispatchQueue.main.async {
-                self.image = placeHolder
-            }
-        }
-        func setImage(_ downloadedImage: UIImage) {
-            DispatchQueue.main.async {
-                self.image = downloadedImage
-            }
-        }
         self.image = nil
         let request = LoadAPIRequest(imageURL: imageURL)
         let resource = Resource<Data>(get: request)
         Dependencies.enviroment.session.download(resource) { imageData, _ in
-            guard let data = imageData,
-                let image = UIImage(data: data) else {
-                setPlaceHolder()
-                    return
+            let image = imageData.map{UIImage.init(data: $0)}
```

Comment:
[nit] If you prefer the use of `$0` like you did here (as opposed to my use of referencing the function directly), then you don't really need the explicit `.init`
```suggestion
            let image = imageData.map{ UIImage(data: $0) }
```

Reply:
How does the referencing function work exactly? 

(Also this code will be deleted soon by the way 😛 )

Reply:
So,

* `map` is a function on a type `SomeWrapper<T>`  (in the present case, it's `Optional.map` so the wrapper is the `Optional` type) that takes a closure `(T) -> U` and returns a `SomeWrapper<U>`
* So in the present case, `imageData.map` is called on `imageData: Data?` (aka `Optional<Data>`), and takes a closure whose signature is `Data -> UIImage`
* You can provide an actual closure body to map, like you did here with `{ UIImage(data: $0) }` for this parameter of `map`. But you can also provide a function that has the same signature. In Swift, functions and closures are interchangeable (as long as they have the same signature of course)
* So if you have a `func makeImage(_ data: Data) -> UIImage` function that already does what you want, you can do `imageData.map(makeImage)` instead of passing a closure literal to `map`, because `makeImage` is a reference to that function and the type of this `makeImage` variable is of the type `Data -> UIImage` already.
* It also happens that the `init` method, even if a special type of function, is also a function, i.e. the `init(data: Data)` initializer on `UIImage` is of type `(Data) -> UIImage` too. But because `init` is a special case of function, when you want to reference it as a function you need to explicitly use the `UIImage.init` name – even though when you call it yourself you can drop the `.init` and just use `UIImage(data: someData)`